crates/services/gas_price_service/src/fuel_gas_price_updater/da_source_adapter/service.rs

```diff
@@ -0,0 +1,130 @@
+use crate::fuel_gas_price_updater::{
+    da_source_adapter::POLLING_INTERVAL_MS,
+    DaBlockCosts,
+};
+use fuel_core_services::{
+    RunnableService,
+    RunnableTask,
+    ServiceRunner,
+    StateWatcher,
+};
+use std::{
+    collections::HashSet,
+    time::Duration,
+};
+use tokio::{
+    sync::mpsc::Sender,
+    time::{
+        interval,
+        Interval,
+    },
+};
+
+pub use anyhow::Result;
+
+/// This struct houses the shared_state, polling interval
+/// and a source, which does the actual fetching of the data
+pub struct DaBlockCostsService<Source>
+where
+    Source: DaBlockCostsSource,
+{
+    poll_interval: Interval,
+    source: Source,
+    sender: Sender<DaBlockCosts>,
+    cache: HashSet<DaBlockCosts>,
```

Comment:
It would be nice to clean it up periodically:) Or even not to use a cache. We can just track the last range/da height of the bundle

Reply:
this will be omitted anyway in the next pr when we restructure the gas price service